cmd/gapit/replace_resource.go

```diff
@@ -90,36 +94,100 @@ func (verb *replaceResourceVerb) Run(ctx context.Context, flags flag.FlagSet) er
 		verb.At = int(boxedCapture.(*service.Capture).NumCommands) - 1
 	}
 
-	for _, types := range resources.GetTypes() {
-		if types.Type == api.ResourceType_ShaderResource {
-			var matchedResource *service.Resource
-			for _, v := range types.GetResources() {
-				if strings.Contains(v.GetHandle(), verb.Handle) {
-					if matchedResource != nil {
-						return fmt.Errorf("Multiple resources matched: %s, %s", matchedResource.GetHandle(), v.GetHandle())
-					}
-					matchedResource = v
-				}
-			}
-			resourcePath := capture.Command(uint64(verb.At)).ResourceAfter(matchedResource.ID)
-			newResourceBytes, err := ioutil.ReadFile(verb.ResourcePath)
+	var resourcePath *path.Any
+	var resourceData interface{}
+
+	switch {
+	case verb.Handle != "":
+		matchedResource, err := resources.FindSingle(func(t api.ResourceType, r service.Resource) bool {
+			return strings.Contains(r.GetHandle(), verb.Handle)
```

Comment:
I think you need to check type here too?

Reply:
Done